src/pbf/mod.rs

```diff
@@ -136,16 +137,25 @@ fn decode_dense_nodes(
         // last_* start off 0
         let id = ids[index] + last_id;
         last_id = id;
-        let lat = lats[index] + last_lat;
-        last_lat = lat;
-        let lon = lons[index] + last_lon;
-        last_lon = lon;
 
-        // FIXME lat/lon here seem to be missing the last digit compared to the website
-        let lat = lat_offset + (granularity * lat);
-        let lat = 0.000000001 * (lat as f32);
-        let lon = lon_offset + (granularity * lon);
-        let lon = 0.000000001 * (lon as f32);
+        let raw_lat = i32::try_from(lats[index] + last_raw_lat as i64)
+            .expect("raw_lat was larger than the OSM precision allows");
```

Comment:
Note this is potentially a new crash. Do you think it's reasonable?

Reply:
In particular, I could see this being problematic for people using pbf format for their own data, outside  OpenStreetMap.org, who want to adopt their own custom more precise precision model.

I guess that it was a very small use case, and that it would be reasonable to explode here for now and re-visit this if anyone ever actually wants to use it this way.

Reply:
While writing this pdf handling code, I saw lots of ways to make invalid PBF file, so it doesn't have to support everything

Reply:
I've been playing around with this library for years, and learned a lot more about Rust. I should go through it and make it return `Result` more often. So I'm not happy with it outright panicing, but I'm willing to accept it because lots of parts of this library need to be properly failing.

Reply:
I've seen[fallible_streaming_iterator](https://docs.rs/fallible-streaming-iterator/0.1.9/fallible_streaming_iterator/)  used in similar processing libraries - it might make sense to use it in osmio too.